arrow-annotations/src/main/java/arrow/optics.kt

```diff
@@ -18,3 +18,7 @@ annotation class isos
 @Retention(SOURCE)
 @Target(CLASS)
 annotation class optionals
+
+@Retention(SOURCE)
+@Target(CLASS)
+annotation class bounded
```

Comment:
I'm not sure about this name. It refers to implementation details and doesn't describe what would be generated from it. Maybe `@setters`, `@boundSetters` or `@syntaxSetters`?

Reply:
Perhaps we could name it `@syntax` for now. Since will be enabled by importing `arrow.optics.syntax.*`.

In #605 we'll combine all code gens by one annotation with fine grained control if needed. I hope to implement this over the weekend so it might even be in before next release.